sys/fmt/fmt.c

```diff
@@ -470,14 +470,13 @@ uint32_t scn_u32_dec(const char *str, size_t n)
     uint32_t res = 0;
 
     while (n--) {
-        char c = *str++;
-        if (!fmt_is_digit(c)) {
+        unsigned char c = *str++;
+        unsigned d = c - '0';
```

Comment:
At least in C99, char literals are of type `int`, so this does not in fact make use of unsigned integer overflows, but instead unleashes undefined behavior.

Reply:
many thanks for pointing that out.

Reply:
I tried to trigger a UB sanatizer that and i could not.
Found out it is not UB but a promotion issue (-> it is well defined to go wrong since if `c` is `unsigned char` it may be promoted to `int` and the literal being `int` is just the icing to trigger that).

I also ran the test before I uploaded this PR but the `char` was a addition to the code after that.